iam.tf

```diff
@@ -318,17 +488,586 @@ data "aws_iam_policy_document" "doublecloud_permissions" {
+      "ec2:Describe*",
+    ]
+    resources = ["*"]
+  }
+  statement {
+    sid    = "AmazonEKSCNIPolicyModifyDeleteV5"
+    effect = "Allow"
+    actions = [
+      "ec2:DeleteNetworkInterface",
+      "ec2:DetachNetworkInterface",
+      "ec2:ModifyNetworkInterfaceAttribute",
+      "ec2:UnassignPrivateIpAddresses"
+    ]
+    resources = ["*"]
+    condition {
+      test     = "StringLike"
+      variable = "aws:ResourceTag/cluster.k8s.amazonaws.com/name"
+      values   = ["DoubleCloud-Airflow-*"]
+    }
+  }
+  statement {
+    sid       = "AmazonEKSCNIPolicyTagsV5"
+    effect    = "Allow"
+    actions   = ["ec2:CreateTags"]
+    resources = ["arn:aws:ec2:*:*:network-interface/*"]
+  }
+  statement {
+    sid    = "AmazonEKSWorkerNodePolicyV3"
+    effect = "Allow"
+    actions = [
+      "ec2:Describe*",
+      "eks-auth:AssumeRoleForPodIdentity"
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_policy" "doublecloud_permission_boundary_eks_node_platform" {
+  name = local.policy_names.permission_boundary_eks_node_platform
+  path = "/DoubleCloud/"
+
+  policy = data.aws_iam_policy_document.doublecloud_permission_boundary_eks_node_platform.json
+}
+
+data "aws_iam_policy_document" "doublecloud_permission_boundary_eks_node_platform" {
+  version = "2012-10-17"
+
+  statement {
+    sid    = "AmazonEC2ContainerRegistryReadOnlyV3"
+    effect = "Allow"
+    actions = [
+      "ecr:BatchCheck*",
+      "ecr:BatchGet*",
+      "ecr:Describe*",
+      "ecr:Get*",
+      "ecr:List*",
+    ]
+    resources = ["*"]
+  }
+  statement {
+    sid    = "AmazonEKSCNIPolicyCreateDescribeV5"
+    effect = "Allow"
+    actions = [
+      "ec2:AssignPrivateIpAddresses",
+      "ec2:AssignIpv6Addresses",
+      "ec2:AttachNetworkInterface",
+      "ec2:CreateNetworkInterface",
+      "ec2:Describe*",
+    ]
+    resources = ["*"]
+  }
+  statement {
+    sid    = "AmazonEKSCNIPolicyModifyDeleteV5"
+    effect = "Allow"
+    actions = [
+      "ec2:DeleteNetworkInterface",
+      "ec2:DetachNetworkInterface",
+      "ec2:ModifyNetworkInterfaceAttribute",
+      "ec2:UnassignPrivateIpAddresses"
+    ]
+    resources = ["*"]
+    condition {
+      test     = "StringLike"
+      variable = "aws:ResourceTag/cluster.k8s.amazonaws.com/name"
+      values   = ["DoubleCloud-Airflow-*"]
+    }
+  }
+  statement {
+    sid       = "AmazonEKSCNIPolicyTagsV5"
+    effect    = "Allow"
+    actions   = ["ec2:CreateTags"]
+    resources = ["arn:aws:ec2:*:*:network-interface/*"]
+  }
+  statement {
+    sid    = "AmazonEKSWorkerNodePolicyV3"
+    effect = "Allow"
+    actions = [
+      "ec2:Describe*",
+      "eks-auth:AssumeRoleForPodIdentity"
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    sid    = "AmazonEBSCSIDriverPolicyEC2CreateDescribeV2"
+    effect = "Allow"
+    actions = [
+      "ec2:CreateSnapshot",
+      "ec2:AttachVolume",
+      "ec2:DetachVolume",
+      "ec2:ModifyVolume",
+      "ec2:Describe*",
+    ]
+    resources = ["*"]
```

Comment:
This is not limited by permission boundaries which is required as it's allows AttachVolume and CreateSnampshot.

Reply:
I do not follow? This is the Permission Boundary for the EKS node.